src/buildblock/TimeFrameDefinitions.cxx

```diff
@@ -275,4 +275,15 @@ set_time_frame(const int frame_num, const double start, const double end)
     frame_times.at(frame_num-1).second = end;
 }
 
+bool TimeFrameDefinitions::operator == (const TimeFrameDefinitions &t) const {
+    bool is_identical=0;
+    
+    for (int frame=0;frame<frame_times.size(); frame++){
+        
+        is_identical=(abs(frame_times.at(frame).first - t.frame_times.at(frame).first) <= 10e-6) &&
+                      (abs(frame_times.at(frame).second - t.frame_times.at(frame).second) <= 10e-6);
+    }
+    return is_identical;
```

Comment:
```suggestion
    for (int frame=0;frame<frame_times.size(); frame++){
        
        const bool is_identical=(abs(frame_times.at(frame).first - t.frame_times.at(frame).first) <= 10e-5) &&
                      (abs(frame_times.at(frame).second - t.frame_times.at(frame).second) <= 10e-5);
        if (!is_identical)
          return false;
    }
    return true;
```
This was effectively only testing for the last frame. Now should be ok I believe.

Also relaxed the numerical error a bit.